store/rootmulti/store.go

```diff
@@ -321,6 +321,14 @@ func (rs *Store) CacheMultiStoreWithVersion(version int64) (types.CacheMultiStor
 	for key, store := range rs.stores {
 		switch store.GetStoreType() {
 		case types.StoreTypeIAVL:
+			// If the store is wrapped with an inter-block cache, we must first unwrap
```

Comment:
Since this is the main fix here, any test to add to actually test this?

Reply:
Test was added in the implementation-side (`types/`).